tests/test_alert_actions.py

```diff
@@ -0,0 +1,242 @@
+            "user_id": _MOCK_ALERT_DETAIL_RESPONSE["alerts"][0]["actorId"],
+        }
+        assert_successful_summary(connector, expected_summary)
+
+    def test_handle_action_when_search_alerts_calls_search(self, mock_py42_client):
+        param = {
+            "username": "gilbert.lequiche@finance.biz",
+            "start_date": "2031-01-01",
+            "end_date": "2031-06-01",
+            "alert_state": "OPEN",
+        }
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        mock_py42_client.alerts.search.assert_called_once()
+
+    def test_handle_action_when_search_alerts_and_all_params_missing_sets_error_message(
+        self, mock_py42_client
+    ):
+        param = {}
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        assert_fail(connector)
+
+    def test_handle_action_when_search_alerts_and_date_range_incomplete_sets_error_message(
+        self, mock_py42_client
+    ):
+        param = {"start_date": "2031-01-01"}
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        assert_fail(connector)
+
+    def test_handle_action_when_search_alerts_and_date_format_invalid_sets_error_message(
+        self, mock_py42_client
+    ):
+        param = {
+            "start_date": "2031-31-12",
+            "end_date": "2032-31-01",
+        }
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        assert_fail(connector)
+
+    def test_handle_action_when_search_alerts_adds_response_items_to_data(
+        self, mock_py42_with_search_alerts
+    ):
+        param = {"username": "barney.frankenberry@chocula.com"}
+        connector = _create_search_alerts_connector(mock_py42_with_search_alerts)
+        connector.handle_action(param)
+        assert_successful_single_data(connector, _MOCK_SEARCH_ALERTS_LIST_RESPONSE)
+
+    def test_handle_action_when_search_alerts_adds_summary(
+        self, mock_py42_with_search_alerts
+    ):
+        param = {"username": "barney.frankenberry@chocula.com"}
+        connector = _create_search_alerts_connector(mock_py42_with_search_alerts)
+        connector.handle_action(param)
+        assert_successful_summary(
+            connector, {"total_count": _MOCK_SEARCH_ALERTS_LIST_RESPONSE["totalCount"]}
+        )
+
+    def test_handle_action_when_search_alerts_and_no_date_range_provided_defaults_to_last_30_days(
+        self, mock_py42_client
+    ):
+        param = {"username": "barney.frankenberry@chocula.com"}
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        actual_query = mock_py42_client.alerts.search.call_args[0][0]
+        # This is not a great assertion...
+        assert len(actual_query._filter_group_list) == 2
```

Comment:
Wasn't able to figure out a clean way to assert that a default "last 30 days" `DateObserved` filter is applied when no date range params are provided. Tried one way involving accessing internal members on the `AlertsQuery` object but it got ugly fast...

Any ideas on how/whether to improve?

Reply:
It's been a while, but I think you could convert it to a dict and read it that way via `dict(actual_query)`.  If I remember right, that object has a special overload that makes it get converted to its JSON representation when you do that.

Reply:
As far as the assertion goes, I typically create an expected date (one 30 days back in this case) and assert that the actual date is equivalent (like subtract them, take absolute value, assert `< 0.01` or something similar).

As far as accessing the value, you can call `str()` on the query - that convert it to JSON and you can parse a `dict` out of it using `json.loads(json_str)` and then access the value that way